### ISSUE TITLE
Any plan to support oauth2-rs futures-03?

### ISSUE THREAD

**Issue by Folyd** [ramosbugs/openidconnect-rs#5]
This crate is great! While `oauth2-rs` support **async/await** recently with `futures-03` feature flag, I just wonder can we ship those features in the next release? 
```toml
[dependencies]
oauth2 = { version = "=3.0.0-alpha.6", features = ["futures-01"] }
```

**Comment by ramosbugs** [comment]
yup, I'd like to do something similar in this crate, but I probably won't have time for a while. I would welcome a PR to do this though!

**Comment by Folyd** [comment]
> yup, I'd like to do something similar in this crate, but I probably won't have time for a while. I would welcome a PR to do this though!

Ok, I'd love to PR.

**Comment by Folyd** [comment]
BTW, I think we should upgrade the `ring` crate to version `0.16.9`. I tried to upgrade but failed.  😢

**Comment by Folyd** [comment]
Great! There is a PR for upgrading `ring` already. https://github.com/ramosbugs/openidconnect-rs/pull/4

**Comment by Folyd** [comment]
@ramosbugs #6 I just submit a PR to upgrade future-0.3 and bump to edition 2018. Please review it. Thanks :)

**Comment by ramosbugs** [comment]
> Great! There is a PR for upgrading `ring` already. #4

Looks like Rocket is still pinned to 0.14, unfortunately: https://crates.io/crates/rocket_http (via `cookie` and `rustls`)... and https://github.com/briansmith/ring/pull/849 still hasn't been merged.

**Comment by ramosbugs** [comment]
Implemented in #6
